File: projekat-eo/src/main/java/com/spring/projekateo/model/ExamPeriod.java
package com.spring.projekateo.model;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name = "exam_period")
public class ExamPeriod {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="exam_period_id",unique = true, nullable = false)
	private Integer id;
	
	@Column(name = "name", nullable = false)
	private String name;
	
	@Column(name = "startDate", nullable = false)
	private Date startDate;
	
	@Column(name = "endDate", nullable = false)
	private Date endDate;
	
	@OneToMany(cascade = CascadeType.REFRESH, fetch = FetchType.LAZY, mappedBy = "period")
	private Set<Exam> exams = new HashSet<Exam>();
	
	public ExamPeriod() {
		
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Date getStartDate() {
		return startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public Set<Exam> getExams() {
		return exams;
	}

	public void setExams(Set<Exam> exams) {
		this.exams = exams;
	}

}

File: projekat-eo-angular/angularApp/src/app/model/examPeriod.ts

export class ExamPeriod implements ExamPeriodInterface{
	public id: number;
	public startDate: Date;
	public endDate: Date;
	public name: String;
	public active: boolean;
		
	constructor(examPeriodCfg:ExamPeriodInterface)
	{	
		this.id = examPeriodCfg.id;
		this.startDate = examPeriodCfg.startDate;
		this.endDate = examPeriodCfg.endDate;	
		this.name = examPeriodCfg.name;
		this.active = examPeriodCfg.active;
	}
}

interface ExamPeriodInterface{
	id?: number;
	startDate: Date;
	endDate: Date;
    name: String;
	active: boolean;
}